src/main/java/tutorial/User.java

```diff
@@ -15,11 +22,15 @@ public class User {
      * @param id the User's id as assigned when the User is created
      */
 
-    public User(String displayName, int id) {
+    public User(String displayName, int id, String username, String password) {
         this.displayName = displayName;
-        this.id = id; //need to determine how we will be creating id and ensuring it does not already exist //
```

Comment:
we have chaned to make username the idential compoent, where we will be checking no existed username will be created in CreateAccountPage